TST: test_roundtrip_high_D fails in numpy dev job

@Cadair and @astrofrog , looks like your FITS compression test from #14252 is not compatible with numpy dev?

Example log: https://github.com/astropy/astropy/actions/runs/3972188461/jobs/6809835077

```
_ test_roundtrip_high_D[GZIP_1 {'qlevel': 5, 'qmethod': -1} <f4-shape4-tile_dim4] _

numpy_rng = Generator(PCG64) at 0x7FBE195AC740, compression_type = 'GZIP_1'
compression_param = {'qlevel': 5, 'qmethod': -1}
tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_roundtrip_high_D_GZIP_1__54')
dtype = '<f4', shape = [2, 3, 4, 5], tile_dim = [5, 5, 1, 1]

    @pytest.mark.parametrize(
        ("shape", "tile_dim"),
        (
            ([10, 10], [5, 5]),  # something for HCOMPRESS
            ([5, 5, 5], [5, 5, 5]),
            # ([5, 5, 5], [5, 5, 1]),  # something for HCOMPRESS
            ([10, 15, 20], [5, 5, 5]),
            ([10, 5, 12], [5, 5, 5]),
            # TODO: There's a stupid bit of code in CompImageHDU which stops this working.
            # ([2, 3, 4, 5], [1, 1, 2, 3]),
            ([2, 3, 4, 5], [5, 5, 1, 1]),
        ),
    )
    def test_roundtrip_high_D(
        numpy_rng, compression_type, compression_param, tmp_path, dtype, shape, tile_dim
    ):
        if compression_type == "HCOMPRESS_1" and (
            # We don't have at least a 2D image
            len(shape) < 2
            or
            # We don't have 2D tiles
            np.count_nonzero(np.array(tile_dim) != 1) != 2
            or
            # TODO: The following restrictions can be lifted with some extra work.
            # The tile is not the first two dimensions of the data
            tile_dim[0] == 1
            or tile_dim[1] == 1
            or
            # The tile dimensions not an integer multiple of the array dims
            np.count_nonzero(np.array(shape[:2]) % tile_dim[:2]) != 0
        ):
            pytest.xfail("HCOMPRESS requires 2D tiles.")
        random = numpy_rng.uniform(high=255, size=shape)
        # Set first value to be exactly zero as zero values require special treatment
        # for SUBTRACTIVE_DITHER_2
        random.ravel()[0] = 0.0
        original_data = random.astype(dtype)
    
        dtype_sanitizer = {
            ">": "big",
            "<": "little",
            "=": "native",
        }
        filename = (
            tmp_path / f"{compression_type}_{dtype[1:]}_{dtype_sanitizer[dtype[0]]}.fits"
        )
    
        param = fitsio_param_to_astropy_param(compression_param)
        hdu = fits.CompImageHDU(
            data=original_data,
            compression_type=compression_type,
            tile_size=tile_dim,
            **param,
        )
        hdu.writeto(filename)
    
        atol = 0
        if compression_param.get("qmethod", None) is not None:
            # This is a horrific hack We are comparing quantized data to unquantized
            # data here, so there can be pretty large differences.  What this test
            # is really checking for is arrays which are *completely* different,
            # which would indicate the compression has not worked.
            atol = 17
    
        with fits.open(filename) as hdul:
            a = hdul[1].data
>           np.testing.assert_allclose(original_data, hdul[1].data, atol=atol)

.../astropy/io/fits/_tiled_compression/tests/test_tiled_compression.py:135: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<function assert_allclose.<locals>.compare at 0x7fbe19a0aa20>, array([[[[  0.       ,  43.990242 , 102.28794  , 156.8... 263.14105  ],
         [ 37.59158  ,  37.59158  , 187.95789  , 112.774734 ,
          150.36632  ]]]], dtype=float32))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=1e-07, atol=17', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Not equal to tolerance rtol=1e-07, atol=17
E           
E           Mismatched elements: 2 / 120 (1.67%)
E           Max absolute difference: 17.739536
E           Max relative difference: 0.47190186
E            x: array([[[[  0.      ,  43.990242, 102.28794 , 156.83925 , 182.48734 ],
E                    [225.95378 ,  68.28684 ,   6.088278, 106.47705 ,  18.340439],
E                    [227.78911 ,  59.23316 , 189.6015  , 139.42618 , 128.34538 ],...
E            y: array([[[[  0.      ,  45.535294,  98.659805, 159.37354 , 182.14117 ],
E                    [227.67647 ,  68.30294 ,   7.589216, 106.24902 ,  15.178432],
E                    [227.67647 ,  60.713726, 189.7304  , 136.60588 , 129.0[1666]

```

--- a/astropy/io/fits/_tiled_compression/tests/conftest.py
+++ b/astropy/io/fits/_tiled_compression/tests/conftest.py
@@ -107,4 +107,4 @@ def dtype(comp_param_dtype):
 
 @pytest.fixture(scope="session")
 def numpy_rng():
-    return np.random.default_rng()
+    return np.random.default_rng(0)

